pkg/interface/src/views/apps/publish/components/lib/Sidebar.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { Component } from "react";
+import { Box, Text, Col } from "@tlon/indigo-react";
+import { Link } from "react-router-dom";
+import SidebarInvite from "~/views/components/SidebarInvite";
+import { Welcome } from "./Welcome";
+import { GroupItem } from "./GroupItem";
+import { alphabetiseAssociations } from "~/logic/lib/util";
+
+export function Sidebar(props: any) {
+  const sidebarInvites = !(props.invites && props.invites["/publish"])
+    ? null
+    : Object.keys(props.invites["/publish"]).map((uid) => {
+        return (
+          <SidebarInvite
+            key={uid}
+            invite={props.invites["/publish"][uid]}
+            onAccept={() => props.api.invite.accept("/publish", uid)}
+            onDecline={() => props.api.invite.decline("/publish", uid)}
+          />
+        );
+      });
+  const associations =
+    props.associations && "contacts" in props.associations
+      ? alphabetiseAssociations(props.associations.contacts)
+      : {};
+
+  const notebooks = {};
+  Object.keys(props.notebooks).map((host) => {
+    Object.keys(props.notebooks[host]).map((notebook) => {
+      const title = `${host}/${notebook}`;
+      notebooks[title] = props.notebooks[host][notebook];
+    });
+  });
+
+  const groupedNotebooks = {};
+  Object.keys(notebooks).map((book) => {
+    const path = notebooks[book]["subscribers-group-path"]
+      ? notebooks[book]["subscribers-group-path"]
+      : book;
+    if (path in associations) {
+      if (groupedNotebooks[path]) {
+        const array = groupedNotebooks[path];
+        array.push(book);
+        groupedNotebooks[path] = array;
+      } else {
+        groupedNotebooks[path] = [book];
+      }
+    } else {
+      if (groupedNotebooks["/~/"]) {
+        const array = groupedNotebooks["/~/"];
+        array.push(book);
+        groupedNotebooks["/~/"] = array;
+      } else {
+        groupedNotebooks["/~/"] = [book];
+      }
+    }
+  });
+
+  const selectedGroups = props.selectedGroups ? props.selectedGroups : [];
```

Comment:
selected groups logic was pulled out in #3231, this, the filter below, and "selectedGroups" in the following conditional statements can be discarded

Reply:
line 59! also L74, L92.